src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/Service/NavigationService.show.cs

```diff
@@ -70,85 +70,95 @@ public virtual async Task ShowAsync<TViewModel>()
         }
 
 
-        private Func<Page, Task> BuildShowAsyncFunc(
-            Page page)
+        public virtual async Task ShowModalAsync<TViewModel>()
+
+            where TViewModel : class, IViewModel
         {
-            if (RootPage is MasterDetailPage &&
-                page is DetailPage)
-            {
-                return SetDetailAsync;
-            }
-            else if (RootPage is TabbedPage &&
-                page is TabPage)
-            {
-                return SetTabAsync;
-            }
-            else return PushAsync;
-        }
+            ThrowIfNotRegistered<TViewModel>();
 
 
-        private async Task PushAsync(
-            Page page)
-        {
-            if (Navigation == null)
-                return;
+            var viewModelInstance =
+                await InitializeViewModelAsync<TViewModel>()
+                .ConfigureAwait(false);
 
-            await Device.InvokeOnMainThreadAsync(() =>
-            {
-                Navigation.PushAsync(
-                    page,
-                    animated: Configuration.UseAnimations);
-            });
+            var page = CreateView<TViewModel>(
+                viewModelInstance);
+
+
+            await PushModalAsync(page);
         }
 
-        private async Task SetDetailAsync(
-            Page page)
+
+        public virtual async Task ShowModalAsync<TViewModel, TData>(
+            TData data)
+
+            where TViewModel : class, IViewModel<TData>
         {
-            if (!(Application.Current.MainPage is MasterDetailPage masterDetailPage))
-                return;
+            ThrowIfNotRegistered<TViewModel>();
 
 
-            await Device.InvokeOnMainThreadAsync(() =>
-            {
-                masterDetailPage.Detail = new NavigationPage(page);
-                masterDetailPage.IsPresented = !Configuration.MasterDetailConfig.HideMenuOnPageSwitch;
-            });
+            var viewModelInstance =
+                await InitializeViewModelAsync<TViewModel, TData>(data)
+                .ConfigureAwait(false);
+
+            var page = CreateView<TViewModel>(
+                viewModelInstance);
+
+
+            await PushModalAsync(page);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5598919)